#!/usr/bin/python

PACKAGE='asv_control'
NODE='dynamic_position'
CONFIG='PoseController'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("surge_Kp", double_t, 0, "X Error PID Proportional Gain" , 0.0, -1000.0, 1000.0)
gen.add("surge_Ki", double_t, 0, "X Error PID Integral Gain"     , 0.0, -1000.0, 1000.0)
gen.add("surge_Kd", double_t, 0, "X Error PID Derivative Gain"   , 0.0, -1000.0, 1000.0)
gen.add("surge_max", double_t, 0, "Maximum X Absolute Output"    , 80.0, 0, 80.0)
gen.add("surge_imax", double_t, 0, "Maximum X Integrator"        , 40.0, 0, 40.0)

gen.add("sway_Kp", double_t, 0, "Y Error PID Proportional Gain" , 0.0, -1000.0, 1000.0)
gen.add("sway_Ki", double_t, 0, "Y Error PID Integral Gain"     , 0.0, -1000.0, 1000.0)
gen.add("sway_Kd", double_t, 0, "Y Error PID Derivative Gain"   , 0.0, -1000.0, 1000.0)
gen.add("sway_max", double_t, 0, "Maximum Y Output"             , 20.0, 0.0, 20.0)
gen.add("sway_imax", double_t, 0, "Maximum Y Integrator"        , 10.0, 0.0, 10.0)

gen.add("yaw_Kp", double_t, 0, "Yaw PID Proportional Gain"  , 0.0, -1000.0, 1000.0)
gen.add("yaw_Ki", double_t, 0, "Yaw PID Integral Gain"      , 0.0, -1000.0, 1000.0)
gen.add("yaw_Kd", double_t, 0, "Yaw PID Derivative Gain"    , 0.0, -1000.0, 1000.0)
gen.add("yaw_max", double_t, 0, "Maximum N Absolute Output" , 20.0, 0.0, 20.0)
gen.add("yaw_imax", double_t, 0, "Maximum N Integrator"     , 10.0, 0.0, 10.0)


exit(gen.generate(PACKAGE, NODE, CONFIG))
